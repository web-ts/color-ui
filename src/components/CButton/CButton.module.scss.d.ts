declare const classNames: {
  readonly CButton: "CButton";
  readonly CButton_solid: "CButton_solid";
  readonly CButton_outline: "CButton_outline";
  readonly CButton_text: "CButton_text";
  readonly CButton_xs: "CButton_xs";
  readonly CButton_sm: "CButton_sm";
  readonly CButton_md: "CButton_md";
  readonly CButton_lg: "CButton_lg";
  readonly CButton_xl: "CButton_xl";
};
export = classNames;
