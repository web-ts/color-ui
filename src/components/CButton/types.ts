import { ElementColor } from "@/types/colors";
import { ElementSize } from "@/types/sizes";
import { JSX } from "solid-js";

export type ButtonProps = {
  children?: JSX.Element | JSX.Element[];
  size?: ElementSize;
  color?: ElementColor;
  variant?: "solid" | "outline" | "text";
  disabled?: boolean;
  loading?: boolean;
  href?: string;
  target?: string;
  replace?: boolean;
  routerLink?: boolean;
  noScroll?: boolean;
  state?: unknown;
  inactiveClass?: string;
  activeClass?: string;
  end?: boolean;
} & JSX.ButtonHTMLAttributes<HTMLButtonElement>;
