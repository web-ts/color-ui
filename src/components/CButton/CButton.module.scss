.CButton {
  cursor: pointer;
  transition: background-color var(--c-default-transition),
    border var(--c-default-transition);
  border-radius: var(--c-rounded);

  &:focus-visible {
    box-shadow: 0 0 0 3px rgba(var(--color-button-base), 0.3);
  }
  &_solid {
    background-color: rgb(var(--color-button-base));
    border: 1px solid rgb(var(--color-button-base));
    color: rgb(var(--color-button-base-text));
    &:hover {
      background-color: rgb(var(--color-button-light));
      border: 1px solid rgb(var(--color-button-light));
    }
    &:active {
      background-color: rgb(var(--color-button-lighter));
      border: 1px solid rgb(var(--color-button-lighter));
    }
  }

  &_outline {
    background-color: rgba(var(--color-button-light), 0.1);
    border: 1px solid rgb(var(--color-button-base));
    color: rgb(var(--color-button-base));
    &:hover {
      background-color: rgba(var(--color-button-light), 0.2);
      border: 1px solid rgb(var(--color-button-light));
    }
    &:active {
      background-color: rgba(var(--color-button-lighter), 0.3);
      border: 1px solid rgb(var(--color-button-lighter));
    }
  }

  &_text {
    background-color: transparent;
    border: 1px solid transparent;
    color: rgb(var(--color-button-base));
    &:hover {
      background-color: rgba(var(--color-button-light), 0.2);
    }
    &:active {
      background-color: rgba(var(--color-button-lighter), 0.3);
    }
  }

  &_xs {
    padding: var(--c-xs-padding) var(--c-md-padding) var(--c-xs-padding)
      var(--c-md-padding);
    font-size: var(--c-xs-font-size);
  }

  &_sm {
    padding: var(--c-sm-padding) var(--c-lg-padding) var(--c-sm-padding)
      var(--c-lg-padding);
    font-size: var(--c-sm-font-size);
  }
  &_md {
    padding: var(--c-md-padding) var(--c-xl-padding) var(--c-md-padding)
      var(--c-xl-padding);
    font-size: var(--c-md-font-size);
  }
  &_lg {
    padding: var(--c-lg-padding) var(--c-2xl-padding) var(--c-lg-padding)
      var(--c-2xl-padding);
    font-size: var(--c-lg-font-size);
  }
  &_xl {
    padding: var(--c-xl-padding) var(--c-3xl-padding) var(--c-xl-padding)
      var(--c-3xl-padding);
    font-size: var(--c-xl-font-size);
  }
}
