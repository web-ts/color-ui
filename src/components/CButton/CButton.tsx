import { Component, createMemo, mergeProps } from "solid-js";
import { Dynamic } from "solid-js/web";

import attribufy from "@/utils/attribufy";
import buildColorStyle from "@/utils/build-color-style";
import { A } from "@solidjs/router";

import SCSS from "./CButton.module.scss";
import { ButtonProps } from "./types";

function attribufyButton(props: ButtonProps) {
  return attribufy(props)("children", "size", "color", "variant", "disabled", "loading", "class", "style");
}

export const CButton: Component<ButtonProps> = (rawProps) => {
  const props = mergeProps(
    {
      color: "primary" as const,
      variant: "solid" as const,
      size: "md" as const
    },
    rawProps
  );
  const classNames = createMemo(() => {
    const classes: string[] = [SCSS.CButton];
    classes.push(SCSS[`CButton_${props.variant}`], SCSS[`CButton_${props.size}`]);

    return `${classes.join(" ")} ${props.class ?? ""}`;
  });

  const style = createMemo(() => {
    return buildColorStyle("color-button", props.color);
  });

  const component = createMemo(() => {
    if (!props.href) return "button";
    if (props.routerLink) return A;
    return "a";
  });

  return (
    <Dynamic component={component()} style={style()} class={classNames()} {...attribufyButton(rawProps)}>
      {rawProps.children}
    </Dynamic>
  );
};
