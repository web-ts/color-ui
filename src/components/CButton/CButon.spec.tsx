import { it, expect, describe } from "vitest";

import { render } from "@solidjs/testing-library";

import { CButton } from "./CButton";
import { Router } from "@solidjs/router";

describe("CButton", () => {
  it("should render a default button", () => {
    const { container } = render(() => <CButton>Test</CButton>);
    expect(container).toMatchSnapshot();
  });

  it("should render all color variants", () => {
    const { container } = render(() => (
      <>
        <CButton color="primary">Test</CButton>
        <CButton color="secondary">Test</CButton>
        <CButton color="success">Test</CButton>
        <CButton color="warning">Test</CButton>
        <CButton color="danger">Test</CButton>
        <CButton color="info">Test</CButton>
      </>
    ));
    expect(container).toMatchSnapshot();
  });

  it("should render all size variants", () => {
    const { container } = render(() => (
      <>
        <CButton size="xs">Test</CButton>
        <CButton size="sm">Test</CButton>
        <CButton size="md">Test</CButton>
        <CButton size="lg">Test</CButton>
        <CButton size="xl">Test</CButton>
      </>
    ));
    expect(container).toMatchSnapshot();
  });

  it("should render all buton variant types", () => {
    const { container } = render(() => (
      <>
        <CButton variant="solid">Test</CButton>
        <CButton variant="outline">Test</CButton>
        <CButton variant="text">Test</CButton>
      </>
    ));
    expect(container).toMatchSnapshot();
  });

  it("should render buttons with all options", () => {
    const { container } = render(() => (
      <>
        <CButton color="primary" size="sm" variant="outline">
          Test
        </CButton>
        <CButton color="secondary" size="lg" variant="text">
          Test
        </CButton>
      </>
    ));

    expect(container).toMatchSnapshot();
  });

  it("should render a button in the loading state", () => {
    const { container } = render(() => <CButton loading>Test</CButton>);

    expect(container).toMatchSnapshot();
  });

  it("should render a button in the loading state with a custom loading icon", () => {
    const { container } = render(() => <CButton loading>Test</CButton>);

    expect(container).toMatchSnapshot();
  });

  it("should render an anchor tag when href is provided", () => {
    const { container } = render(() => (
      <CButton href="https://google.com">Test</CButton>
    ));

    expect(container).toMatchSnapshot();

    const { container: container2 } = render(() => (
      <Router>
        <CButton href="/page" routerLink>
          Test
        </CButton>
      </Router>
    ));

    expect(container2).toMatchSnapshot();
  });
});
