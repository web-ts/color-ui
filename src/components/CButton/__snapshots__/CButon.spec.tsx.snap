// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CButton > should render a button in the loading state 1`] = `
<div>
  <button
    class="_CButton_4ca42b _CButton_solid_4ca42b _CButton_md_4ca42b "
    style="--color-button-base: var(--color-primary-base); --color-button-base-text: var(--color-primary-base-text); --color-button-light: var(--color-primary-light); --color-button-light-text: var(--color-primary-light-text); --color-button-lighter: var(--color-primary-lighter); --color-button-lighter-text: var(--color-primary-lighter-text); --color-button-lightest: var(--color-primary-lightest); --color-button-lightest-text: var(--color-primary-lightest-text); --color-button-dark: var(--color-primary-dark); --color-button-dark-text: var(--color-primary-dark-text); --color-button-darker: var(--color-primary-darker); --color-button-darker-text: var(--color-primary-darker-text); --color-button-darkest: var(--color-primary-darkest); --color-button-darkest-text: var(--color-primary-darkest-text);"
  >
    Test
  </button>
</div>
`;

exports[`CButton > should render a button in the loading state with a custom loading icon 1`] = `
<div>
  <button
    class="_CButton_4ca42b _CButton_solid_4ca42b _CButton_md_4ca42b "
    style="--color-button-base: var(--color-primary-base); --color-button-base-text: var(--color-primary-base-text); --color-button-light: var(--color-primary-light); --color-button-light-text: var(--color-primary-light-text); --color-button-lighter: var(--color-primary-lighter); --color-button-lighter-text: var(--color-primary-lighter-text); --color-button-lightest: var(--color-primary-lightest); --color-button-lightest-text: var(--color-primary-lightest-text); --color-button-dark: var(--color-primary-dark); --color-button-dark-text: var(--color-primary-dark-text); --color-button-darker: var(--color-primary-darker); --color-button-darker-text: var(--color-primary-darker-text); --color-button-darkest: var(--color-primary-darkest); --color-button-darkest-text: var(--color-primary-darkest-text);"
  >
    Test
  </button>
</div>
`;

exports[`CButton > should render a default button 1`] = `
<div>
  <button
    class="_CButton_4ca42b _CButton_solid_4ca42b _CButton_md_4ca42b "
    style="--color-button-base: var(--color-primary-base); --color-button-base-text: var(--color-primary-base-text); --color-button-light: var(--color-primary-light); --color-button-light-text: var(--color-primary-light-text); --color-button-lighter: var(--color-primary-lighter); --color-button-lighter-text: var(--color-primary-lighter-text); --color-button-lightest: var(--color-primary-lightest); --color-button-lightest-text: var(--color-primary-lightest-text); --color-button-dark: var(--color-primary-dark); --color-button-dark-text: var(--color-primary-dark-text); --color-button-darker: var(--color-primary-darker); --color-button-darker-text: var(--color-primary-darker-text); --color-button-darkest: var(--color-primary-darkest); --color-button-darkest-text: var(--color-primary-darkest-text);"
  >
    Test
  </button>
</div>
`;

exports[`CButton > should render all buton variant types 1`] = `
<div>
  <button
    class="_CButton_4ca42b _CButton_solid_4ca42b _CButton_md_4ca42b "
    style="--color-button-base: var(--color-primary-base); --color-button-base-text: var(--color-primary-base-text); --color-button-light: var(--color-primary-light); --color-button-light-text: var(--color-primary-light-text); --color-button-lighter: var(--color-primary-lighter); --color-button-lighter-text: var(--color-primary-lighter-text); --color-button-lightest: var(--color-primary-lightest); --color-button-lightest-text: var(--color-primary-lightest-text); --color-button-dark: var(--color-primary-dark); --color-button-dark-text: var(--color-primary-dark-text); --color-button-darker: var(--color-primary-darker); --color-button-darker-text: var(--color-primary-darker-text); --color-button-darkest: var(--color-primary-darkest); --color-button-darkest-text: var(--color-primary-darkest-text);"
  >
    Test
  </button>
  <button
    class="_CButton_4ca42b _CButton_outline_4ca42b _CButton_md_4ca42b "
    style="--color-button-base: var(--color-primary-base); --color-button-base-text: var(--color-primary-base-text); --color-button-light: var(--color-primary-light); --color-button-light-text: var(--color-primary-light-text); --color-button-lighter: var(--color-primary-lighter); --color-button-lighter-text: var(--color-primary-lighter-text); --color-button-lightest: var(--color-primary-lightest); --color-button-lightest-text: var(--color-primary-lightest-text); --color-button-dark: var(--color-primary-dark); --color-button-dark-text: var(--color-primary-dark-text); --color-button-darker: var(--color-primary-darker); --color-button-darker-text: var(--color-primary-darker-text); --color-button-darkest: var(--color-primary-darkest); --color-button-darkest-text: var(--color-primary-darkest-text);"
  >
    Test
  </button>
  <button
    class="_CButton_4ca42b _CButton_text_4ca42b _CButton_md_4ca42b "
    style="--color-button-base: var(--color-primary-base); --color-button-base-text: var(--color-primary-base-text); --color-button-light: var(--color-primary-light); --color-button-light-text: var(--color-primary-light-text); --color-button-lighter: var(--color-primary-lighter); --color-button-lighter-text: var(--color-primary-lighter-text); --color-button-lightest: var(--color-primary-lightest); --color-button-lightest-text: var(--color-primary-lightest-text); --color-button-dark: var(--color-primary-dark); --color-button-dark-text: var(--color-primary-dark-text); --color-button-darker: var(--color-primary-darker); --color-button-darker-text: var(--color-primary-darker-text); --color-button-darkest: var(--color-primary-darkest); --color-button-darkest-text: var(--color-primary-darkest-text);"
  >
    Test
  </button>
</div>
`;

exports[`CButton > should render all color variants 1`] = `
<div>
  <button
    class="_CButton_4ca42b _CButton_solid_4ca42b _CButton_md_4ca42b "
    style="--color-button-base: var(--color-primary-base); --color-button-base-text: var(--color-primary-base-text); --color-button-light: var(--color-primary-light); --color-button-light-text: var(--color-primary-light-text); --color-button-lighter: var(--color-primary-lighter); --color-button-lighter-text: var(--color-primary-lighter-text); --color-button-lightest: var(--color-primary-lightest); --color-button-lightest-text: var(--color-primary-lightest-text); --color-button-dark: var(--color-primary-dark); --color-button-dark-text: var(--color-primary-dark-text); --color-button-darker: var(--color-primary-darker); --color-button-darker-text: var(--color-primary-darker-text); --color-button-darkest: var(--color-primary-darkest); --color-button-darkest-text: var(--color-primary-darkest-text);"
  >
    Test
  </button>
  <button
    class="_CButton_4ca42b _CButton_solid_4ca42b _CButton_md_4ca42b "
    style="--color-button-base: var(--color-secondary-base); --color-button-base-text: var(--color-secondary-base-text); --color-button-light: var(--color-secondary-light); --color-button-light-text: var(--color-secondary-light-text); --color-button-lighter: var(--color-secondary-lighter); --color-button-lighter-text: var(--color-secondary-lighter-text); --color-button-lightest: var(--color-secondary-lightest); --color-button-lightest-text: var(--color-secondary-lightest-text); --color-button-dark: var(--color-secondary-dark); --color-button-dark-text: var(--color-secondary-dark-text); --color-button-darker: var(--color-secondary-darker); --color-button-darker-text: var(--color-secondary-darker-text); --color-button-darkest: var(--color-secondary-darkest); --color-button-darkest-text: var(--color-secondary-darkest-text);"
  >
    Test
  </button>
  <button
    class="_CButton_4ca42b _CButton_solid_4ca42b _CButton_md_4ca42b "
    style="--color-button-base: var(--color-success-base); --color-button-base-text: var(--color-success-base-text); --color-button-light: var(--color-success-light); --color-button-light-text: var(--color-success-light-text); --color-button-lighter: var(--color-success-lighter); --color-button-lighter-text: var(--color-success-lighter-text); --color-button-lightest: var(--color-success-lightest); --color-button-lightest-text: var(--color-success-lightest-text); --color-button-dark: var(--color-success-dark); --color-button-dark-text: var(--color-success-dark-text); --color-button-darker: var(--color-success-darker); --color-button-darker-text: var(--color-success-darker-text); --color-button-darkest: var(--color-success-darkest); --color-button-darkest-text: var(--color-success-darkest-text);"
  >
    Test
  </button>
  <button
    class="_CButton_4ca42b _CButton_solid_4ca42b _CButton_md_4ca42b "
    style="--color-button-base: var(--color-warning-base); --color-button-base-text: var(--color-warning-base-text); --color-button-light: var(--color-warning-light); --color-button-light-text: var(--color-warning-light-text); --color-button-lighter: var(--color-warning-lighter); --color-button-lighter-text: var(--color-warning-lighter-text); --color-button-lightest: var(--color-warning-lightest); --color-button-lightest-text: var(--color-warning-lightest-text); --color-button-dark: var(--color-warning-dark); --color-button-dark-text: var(--color-warning-dark-text); --color-button-darker: var(--color-warning-darker); --color-button-darker-text: var(--color-warning-darker-text); --color-button-darkest: var(--color-warning-darkest); --color-button-darkest-text: var(--color-warning-darkest-text);"
  >
    Test
  </button>
  <button
    class="_CButton_4ca42b _CButton_solid_4ca42b _CButton_md_4ca42b "
    style="--color-button-base: var(--color-danger-base); --color-button-base-text: var(--color-danger-base-text); --color-button-light: var(--color-danger-light); --color-button-light-text: var(--color-danger-light-text); --color-button-lighter: var(--color-danger-lighter); --color-button-lighter-text: var(--color-danger-lighter-text); --color-button-lightest: var(--color-danger-lightest); --color-button-lightest-text: var(--color-danger-lightest-text); --color-button-dark: var(--color-danger-dark); --color-button-dark-text: var(--color-danger-dark-text); --color-button-darker: var(--color-danger-darker); --color-button-darker-text: var(--color-danger-darker-text); --color-button-darkest: var(--color-danger-darkest); --color-button-darkest-text: var(--color-danger-darkest-text);"
  >
    Test
  </button>
  <button
    class="_CButton_4ca42b _CButton_solid_4ca42b _CButton_md_4ca42b "
    style="--color-button-base: var(--color-info-base); --color-button-base-text: var(--color-info-base-text); --color-button-light: var(--color-info-light); --color-button-light-text: var(--color-info-light-text); --color-button-lighter: var(--color-info-lighter); --color-button-lighter-text: var(--color-info-lighter-text); --color-button-lightest: var(--color-info-lightest); --color-button-lightest-text: var(--color-info-lightest-text); --color-button-dark: var(--color-info-dark); --color-button-dark-text: var(--color-info-dark-text); --color-button-darker: var(--color-info-darker); --color-button-darker-text: var(--color-info-darker-text); --color-button-darkest: var(--color-info-darkest); --color-button-darkest-text: var(--color-info-darkest-text);"
  >
    Test
  </button>
</div>
`;

exports[`CButton > should render all size variants 1`] = `
<div>
  <button
    class="_CButton_4ca42b _CButton_solid_4ca42b _CButton_xs_4ca42b "
    style="--color-button-base: var(--color-primary-base); --color-button-base-text: var(--color-primary-base-text); --color-button-light: var(--color-primary-light); --color-button-light-text: var(--color-primary-light-text); --color-button-lighter: var(--color-primary-lighter); --color-button-lighter-text: var(--color-primary-lighter-text); --color-button-lightest: var(--color-primary-lightest); --color-button-lightest-text: var(--color-primary-lightest-text); --color-button-dark: var(--color-primary-dark); --color-button-dark-text: var(--color-primary-dark-text); --color-button-darker: var(--color-primary-darker); --color-button-darker-text: var(--color-primary-darker-text); --color-button-darkest: var(--color-primary-darkest); --color-button-darkest-text: var(--color-primary-darkest-text);"
  >
    Test
  </button>
  <button
    class="_CButton_4ca42b _CButton_solid_4ca42b _CButton_sm_4ca42b "
    style="--color-button-base: var(--color-primary-base); --color-button-base-text: var(--color-primary-base-text); --color-button-light: var(--color-primary-light); --color-button-light-text: var(--color-primary-light-text); --color-button-lighter: var(--color-primary-lighter); --color-button-lighter-text: var(--color-primary-lighter-text); --color-button-lightest: var(--color-primary-lightest); --color-button-lightest-text: var(--color-primary-lightest-text); --color-button-dark: var(--color-primary-dark); --color-button-dark-text: var(--color-primary-dark-text); --color-button-darker: var(--color-primary-darker); --color-button-darker-text: var(--color-primary-darker-text); --color-button-darkest: var(--color-primary-darkest); --color-button-darkest-text: var(--color-primary-darkest-text);"
  >
    Test
  </button>
  <button
    class="_CButton_4ca42b _CButton_solid_4ca42b _CButton_md_4ca42b "
    style="--color-button-base: var(--color-primary-base); --color-button-base-text: var(--color-primary-base-text); --color-button-light: var(--color-primary-light); --color-button-light-text: var(--color-primary-light-text); --color-button-lighter: var(--color-primary-lighter); --color-button-lighter-text: var(--color-primary-lighter-text); --color-button-lightest: var(--color-primary-lightest); --color-button-lightest-text: var(--color-primary-lightest-text); --color-button-dark: var(--color-primary-dark); --color-button-dark-text: var(--color-primary-dark-text); --color-button-darker: var(--color-primary-darker); --color-button-darker-text: var(--color-primary-darker-text); --color-button-darkest: var(--color-primary-darkest); --color-button-darkest-text: var(--color-primary-darkest-text);"
  >
    Test
  </button>
  <button
    class="_CButton_4ca42b _CButton_solid_4ca42b _CButton_lg_4ca42b "
    style="--color-button-base: var(--color-primary-base); --color-button-base-text: var(--color-primary-base-text); --color-button-light: var(--color-primary-light); --color-button-light-text: var(--color-primary-light-text); --color-button-lighter: var(--color-primary-lighter); --color-button-lighter-text: var(--color-primary-lighter-text); --color-button-lightest: var(--color-primary-lightest); --color-button-lightest-text: var(--color-primary-lightest-text); --color-button-dark: var(--color-primary-dark); --color-button-dark-text: var(--color-primary-dark-text); --color-button-darker: var(--color-primary-darker); --color-button-darker-text: var(--color-primary-darker-text); --color-button-darkest: var(--color-primary-darkest); --color-button-darkest-text: var(--color-primary-darkest-text);"
  >
    Test
  </button>
  <button
    class="_CButton_4ca42b _CButton_solid_4ca42b _CButton_xl_4ca42b "
    style="--color-button-base: var(--color-primary-base); --color-button-base-text: var(--color-primary-base-text); --color-button-light: var(--color-primary-light); --color-button-light-text: var(--color-primary-light-text); --color-button-lighter: var(--color-primary-lighter); --color-button-lighter-text: var(--color-primary-lighter-text); --color-button-lightest: var(--color-primary-lightest); --color-button-lightest-text: var(--color-primary-lightest-text); --color-button-dark: var(--color-primary-dark); --color-button-dark-text: var(--color-primary-dark-text); --color-button-darker: var(--color-primary-darker); --color-button-darker-text: var(--color-primary-darker-text); --color-button-darkest: var(--color-primary-darkest); --color-button-darkest-text: var(--color-primary-darkest-text);"
  >
    Test
  </button>
</div>
`;

exports[`CButton > should render an anchor tag when href is provided 1`] = `
<div>
  <a
    class="_CButton_4ca42b _CButton_solid_4ca42b _CButton_md_4ca42b "
    href="https://google.com"
    style="--color-button-base: var(--color-primary-base); --color-button-base-text: var(--color-primary-base-text); --color-button-light: var(--color-primary-light); --color-button-light-text: var(--color-primary-light-text); --color-button-lighter: var(--color-primary-lighter); --color-button-lighter-text: var(--color-primary-lighter-text); --color-button-lightest: var(--color-primary-lightest); --color-button-lightest-text: var(--color-primary-lightest-text); --color-button-dark: var(--color-primary-dark); --color-button-dark-text: var(--color-primary-dark-text); --color-button-darker: var(--color-primary-darker); --color-button-darker-text: var(--color-primary-darker-text); --color-button-darkest: var(--color-primary-darkest); --color-button-darkest-text: var(--color-primary-darkest-text);"
  >
    Test
  </a>
</div>
`;

exports[`CButton > should render an anchor tag when href is provided 2`] = `
<div>
  <a
    class="_CButton_4ca42b _CButton_solid_4ca42b _CButton_md_4ca42b inactive"
    href="/page"
    link=""
    routerlink="true"
    style="--color-button-base: var(--color-primary-base); --color-button-base-text: var(--color-primary-base-text); --color-button-light: var(--color-primary-light); --color-button-light-text: var(--color-primary-light-text); --color-button-lighter: var(--color-primary-lighter); --color-button-lighter-text: var(--color-primary-lighter-text); --color-button-lightest: var(--color-primary-lightest); --color-button-lightest-text: var(--color-primary-lightest-text); --color-button-dark: var(--color-primary-dark); --color-button-dark-text: var(--color-primary-dark-text); --color-button-darker: var(--color-primary-darker); --color-button-darker-text: var(--color-primary-darker-text); --color-button-darkest: var(--color-primary-darkest); --color-button-darkest-text: var(--color-primary-darkest-text);"
  >
    Test
  </a>
</div>
`;

exports[`CButton > should render buttons with all options 1`] = `
<div>
  <button
    class="_CButton_4ca42b _CButton_outline_4ca42b _CButton_sm_4ca42b "
    style="--color-button-base: var(--color-primary-base); --color-button-base-text: var(--color-primary-base-text); --color-button-light: var(--color-primary-light); --color-button-light-text: var(--color-primary-light-text); --color-button-lighter: var(--color-primary-lighter); --color-button-lighter-text: var(--color-primary-lighter-text); --color-button-lightest: var(--color-primary-lightest); --color-button-lightest-text: var(--color-primary-lightest-text); --color-button-dark: var(--color-primary-dark); --color-button-dark-text: var(--color-primary-dark-text); --color-button-darker: var(--color-primary-darker); --color-button-darker-text: var(--color-primary-darker-text); --color-button-darkest: var(--color-primary-darkest); --color-button-darkest-text: var(--color-primary-darkest-text);"
  >
    Test
  </button>
  <button
    class="_CButton_4ca42b _CButton_text_4ca42b _CButton_lg_4ca42b "
    style="--color-button-base: var(--color-secondary-base); --color-button-base-text: var(--color-secondary-base-text); --color-button-light: var(--color-secondary-light); --color-button-light-text: var(--color-secondary-light-text); --color-button-lighter: var(--color-secondary-lighter); --color-button-lighter-text: var(--color-secondary-lighter-text); --color-button-lightest: var(--color-secondary-lightest); --color-button-lightest-text: var(--color-secondary-lightest-text); --color-button-dark: var(--color-secondary-dark); --color-button-dark-text: var(--color-secondary-dark-text); --color-button-darker: var(--color-secondary-darker); --color-button-darker-text: var(--color-secondary-darker-text); --color-button-darkest: var(--color-secondary-darkest); --color-button-darkest-text: var(--color-secondary-darkest-text);"
  >
    Test
  </button>
</div>
`;
